### PR TITLE
[LLHD] Refactor enum class and move header file locations.

### PR DESCRIPTION
**Pull request by DeepFlyingSky**
Refactor `enum TraceMode` to `enum class TraceMode` to enrich
  the readability. And move header files to avoid re-declare like
  enum TraceFormat.

 Currently, there exists two enum type related to TraceMode.
 `enum TraceMode` in Trace.h and `enum TraceFormat` in llhd-sim.cpp.
 Combine with `enum class TraceMode` .

### CODE REVIEW

**Review by fabianschuiki — APPROVED** [review]
I think this looks great, but let's maybe wait for to see if @maerhart or @rodonisi have any concerns or comments :+1:

File: include/circt/Dialect/LLHD/Simulator/Engine.h [hunk 1]
```diff
@@ -13,6 +13,9 @@
 #ifndef CIRCT_DIALECT_LLHD_SIMULATOR_ENGINE_H
 #define CIRCT_DIALECT_LLHD_SIMULATOR_ENGINE_H
 
+#include "State.h"
```
Comment (maerhart) [inline include/circt/Dialect/LLHD/Simulator/Engine.h:16]: Why do we need this include? There are forward-declarations of `State` and `Instance` below, so shouldn't we be able to get rid of one of them?

**Review by maerhart — APPROVED** [review]
Looks good to me! Thanks for doing all these refactorings!

File: include/circt/Dialect/LLHD/Simulator/Engine.h (hunk 1 above)
Comment (DeepFlyingSky) [reply include/circt/Dialect/LLHD/Simulator/Engine.h:16]: Thanks and I miss it, prefer to remove the declarations of `State` and `Instance`.

**Comment by DeepFlyingSky** [comment]
Thanks for reviews @fabianschuiki  and @maerhart .

**Pull request merged**